src/main/java/frc/team2412/robot/subsystem/ClimbSubsystem.java

```diff
@@ -101,11 +102,13 @@
 
     private double lastUpdatedTime = Timer.getFPGATimestamp();
 
-    public ClimbSubsystem(WPI_TalonFX climbFixedMotor, WPI_TalonFX climbDynamicMotor, DoubleSolenoid climbAngle) {
+    private ClimbSubsystem() {
```

Comment:
Getting rid of the long parameter list from the subsystems constructors is goodness that I'd like to keep somehow. From the [robotbuilder subsystem](https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robotbuilder/writing-code/robotbuilder-writing-subsystem-code.html) and [command-based example code](https://docs.wpilib.org/en/stable/docs/software/commandbased/structuring-command-based-project.html#robotcontainer) subsystems do their own hardware creation instead of having everything passed in. Perhaps `Hardware` can be simplified to just have the CAN ID (and other hardware ID) constants and let the subsystems create the motor or solenoid classes.

Reply:
Having each subsystem do the creation of the motor / solenoid / etc. also ensures it's impossible to pass the same hardware item instance to two different subsystems, which is additional goodness.

Reply:
I agree with your first point, but for your second point, I think subsystems could still try to make separate hardware item instances with the same ID, which, as far as I know, would also be problematic.

Reply:
If the IDs come from the hardware class, I don't think this would be an issue, or at least it wouldn't be any more of an issue than our current setup.

Reply:
the issue isnt the subsystem parameters its the command parameters you only create subsystems once